src/plugins/intel_gpu/src/graph/primitive_inst.cpp

```diff
@@ -429,7 +429,8 @@ void primitive_inst::update_shape() {
             continue;
         }
 
-        if (!get_node().is_type<shape_of>() && !dep->get_node().is_in_shape_of_subgraph()) {
+        if (!get_node().is_type<shape_of>() &&
+            !(dep->get_node().is_in_shape_of_subgraph() && dep->get_node().get_selected_impl()->is_cpu())) {
```

Comment:
In this case, I think we can just replace it with 
!get_node().is_type<shape_of>() && !dep->get_node().get_selected_impl()->is_cpu() 
Because if the dep is cpu, the execution is already finished.

Reply:
Updated! Thanks!